config/routes.rb

```diff
@@ -1,6 +1,14 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  # Route for displaying all posts for a specific user
+  get 'users/:user_id/posts', to: 'posts#index', as: 'user_posts'
 
-  # Defines the root path route ("/")
-  # root "articles#index"
+  # Route for displaying a specific post and its comments
+  get 'users/:user_id/posts/:id', to: 'posts#show', as: 'user_post'
+
+  # Route for displaying all user names
+  get 'users', to: 'users#index'
+
+  # Route for displaying a specific user and their posts
+  get 'users/:id', to: 'users#show', as: 'user'
```

Comment:
- [ ] [optional] I would suggest replacing the routes with the below code. It will make the code more readable and follow best practices.

```ruby
resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end

Reply:
fixed, thanks for the feedback ♻️